azure-tools-common/src/main/java/com/microsoft/azure/maven/function/handlers/runtime/DockerFunctionRuntimeHandler.java

```diff
@@ -79,13 +69,13 @@ protected DockerFunctionRuntimeHandler(Builder builder) {
         final String decryptionKey = generateDecryptionKey();
         return (FunctionApp.DefinitionStages.WithCreate) result
                 .withAppSetting(APP_SETTING_MACHINEKEY_DECRYPTION_KEY, decryptionKey)
-                .withAppSetting(APP_SETTING_WEBSITES_ENABLE_APP_SERVICE_STORAGE, APP_SETTING_WEBSITES_ENABLE_APP_SERVICE_STORAGE_VALUE)
-                .withAppSetting(APP_SETTING_FUNCTION_APP_EDIT_MODE, APP_SETTING_FUNCTION_APP_EDIT_MODE_VALUE);
+                .withAppSetting(APP_SETTING_WEBSITES_ENABLE_APP_SERVICE_STORAGE, APP_SETTING_DISABLE_WEBSITES_APP_SERVICE_STORAGE)
+                .withAppSetting(APP_SETTING_FUNCTION_APP_EDIT_MODE, APP_SETTING_FUNCTION_APP_EDIT_MODE_READONLY);
     }
 
     @Override
     public FunctionApp.Update updateAppRuntime(FunctionApp app) throws AzureExecutionException {
-        final MavenDockerCredentialProvider provider = MavenDockerCredentialProvider.fromMavenSettings(settings, serverId);
+        final IDockerCredentialProvider provider = StringUtils.isNotBlank(this.serverId) ? this.dockerCredentialProvider : null;
```

Comment:
Why we add a `StringUtils.isNotBlank(this.serverId)` check here? As `DockerFunctionRuntimeHandler` should not know the implementation of CredentialProvider

Reply:
Done